### ISSUE TITLE
Scripting support?

### ISSUE THREAD

**Issue by Thorium** [MicrosoftDocs/azure-devops-docs#11059]
If I just created a new VM from ARM-template... how do I register it automatically as an environment?

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 91d0d31f-81ee-c024-db7e-daddbf525f71
* Version Independent ID: 330f1649-386c-d0aa-5f96-b8343a1480d3
* Content: [Environment - Virtual machine resource - Azure Pipelines](https://docs.microsoft.com/en-us/azure/devops/pipelines/process/environments-virtual-machines?view=azure-devops)
* Content Source: [docs/pipelines/process/environments-virtual-machines.md](https://github.com/MicrosoftDocs/azure-devops-docs/blob/master/docs/pipelines/process/environments-virtual-machines.md)
* Product: **devops**
* Technology: **devops-cicd-process**
* GitHub Login: @juliakm
* Microsoft Alias: **jukullam**

**Comment by TimShererWithAquent** [comment]
@Thorium Thanks for your question. While we're looking into changes for the article, consider these resources:

- [Azure DevOps Services](https://azure.microsoft.com/support/devops/) 
- [Azure DevOps Support Bot](https://azuredevopsvirtualagent.azurewebsites.net/) 
- [Azure DevOps on Stack Overflow](https://stackoverflow.com/questions/tagged/azure-devops)

**Comment by juliakm** [comment]
@Thorium I'm checking on this with the product team. Are you installing the VM registration script as part of your ARM template? Since this is an implementation question, you may be able to get a faster answer in our [Developer Community](https://developercommunity.visualstudio.com/search?space=21). If you do that and get an answer, please let us know so that we can update the relevant docs.

**Comment by juliakm** [comment]
@Thorium Yes, it should be possible. You'll need to have the Environment already exist and then use the registration script for a VM within your pipeline. I found this [example blog post](https://ronaldbosma.github.io/blog/2021/03/16/provision-an-azure-vm-in-an-azure-pipelines-environment/) helpful for navigating the steps. Closing this issue for now because, given the implementation complexity, it doesn't seem like a good fit for a documentation addition.
